hazelcast/src/main/java/com/hazelcast/map/impl/operation/QueryOperation.java

```diff
@@ -114,6 +111,41 @@ public void run() throws Exception {
         checkPartitionStateChanges(partitionService, partitionStateVersion);
     }
 
+    private void fullTableScan(Collection<Integer> initialPartitions, GroupProperties groupProperties)
+            throws InterruptedException, ExecutionException {
+        if (pagingPredicate != null) {
+            runParallelForPaging(initialPartitions);
+        } else {
+            boolean parallelEvaluation = groupProperties.QUERY_PREDICATE_PARALLEL_EVALUATION.getBoolean();
+            if (parallelEvaluation) {
+                runParallel(initialPartitions);
+            } else {
+                runSingleThreaded(initialPartitions);
+            }
+        }
+    }
+
+    protected void runSingleThreaded(final Collection<Integer> initialPartitions) {
+        RuntimeException storedException = null;
+        MapServiceContext mapServiceContext = mapService.getMapServiceContext();
+        MapContextQuerySupport querySupport = mapServiceContext.getMapContextQuerySupport();
+        for (Integer partitionId : initialPartitions) {
+            try {
+                Collection<QueryableEntry> entries = querySupport.queryOnPartition(name, predicate, partitionId);
+                if (!addQueryEntriesToResult(entries)) {
+                    break;
+                }
+            } catch (RuntimeException e) {
+                if (storedException == null) {
```

Comment:
It's actually needed - as otherwise some tests were failing. 

Here is why:
- Let's assume I have a map with a (slow) MapLoader. The map has been just created and no operation has been executed against this map.
- Someone sends a new `QueryOperation`.
- The parallel execution sends tasks to every partition and waits for results
- Each task will trigger partition map creation. This trigger content to by loaded from a MapLoader 
- the `future.get()` throws an RetryableException because it's still loading from a MapLoader -> The Exception is propagated to a caller. 
- The important thing is the tasks for all partitions had been already send -> they will trigger initialization MapLoader for all map partitions even when the very first `future.get()` throws an  Exception
- The caller eventually retry the QueryOperation -> at this point all partitions are ready with a content loaded

Things are quite different when the QueryOperation is not offloading tasks to Query Executor:
- It calls `querySupport.queryOnPartition(name, predicate, partitionId);` in a loop for all partitions. - Each call will initialization a given partition -> it triggers mapLoader -> the call throws the RetryableException (because it's loading)
- Now unlike the parallel scenario the Exception is propagated back to a caller and no additional map partition initialization is trigged!
- The caller will eventually retry the QueryOperation -> the 1st iteration will succeed (because the 1st map partition is already loaded at this point), but the 2nd iteration will fail again.

I hope you made it a bit clearer :)